ldap/servers/slapd/connection.c

```diff
@@ -1938,6 +1958,18 @@ connection_threadmain(void *arg)
                      * are bypassing both of those, we set idlesince here
                      */
                     conn->c_idlesince = curtime;
+#ifdef ENABLE_EPOLL
+                    if (conn->c_idle_tfd >= 0) {
+                        /* Reset the idle timer */
+                        slapi_log_err(SLAPI_LOG_DEBUG,
+                                      "handle_pr_read_ready", "resetting idle timer for connection %d to %d\n", conn->c_ci, conn->c_idletimeout);
+                        timerfd_settime(conn->c_idle_tfd, 0,
```

Comment:
If the connection is in CONN_FLAG_CLOSING, then won't this timerfd_settime fail as the event was removed?

Reply:
The itimerspec is attached to the actual fd, not the epoll event struct.  It is reset here to avoid firing while closing.